src/main/java/nextstep/subway/line/domain/Section.java

```diff
@@ -64,20 +70,14 @@ public boolean isMatchDownStation(final Station source) {
         return Objects.equals(downStation, source);
     }
 
-    public void updateUpStation(Station station, int newDistance) {
-        if (this.distance <= newDistance) {
-            throw new RuntimeException("역과 역 사이의 거리보다 좁은 거리를 입력해주세요");
-        }
+    public void updateUpStation(Station station, Distance newDistance) {
         this.upStation = station;
-        this.distance -= newDistance;
+        this.distance.minus(newDistance);
     }
 
-    public void updateDownStation(Station station, int newDistance) {
-        if (this.distance <= newDistance) {
-            throw new RuntimeException("역과 역 사이의 거리보다 좁은 거리를 입력해주세요");
-        }
+    public void updateDownStation(Station station, Distance newDistance) {
         this.downStation = station;
-        this.distance -= newDistance;
+        this.distance.minus(newDistance);
```

Comment:
```suggestion
        this.distance = this.distance.minus(newDistance);
```

값객체라면 새로 할당이 필요하지 않을까요? 🤔

Reply:
네 맞습니다~ 음 이부분을 왜 다시 주입안했을까요 ... 
꼼꼼한 코드 리뷰 감사합니다 ^^